src/shogun/modelselection/ParameterCombination.cpp

```diff
@@ -768,10 +778,11 @@ void CParameterCombination::build_parameter_values_map(
 			if (type.m_ptype==PT_FLOAT64 || type.m_ptype==PT_FLOAT32 ||
 					type.m_ptype==PT_FLOATMAX)
 			{
-				if ((type.m_ctype==CT_SGVECTOR || type.m_ctype==CT_VECTOR))
+				if (type.m_ctype==CT_SGVECTOR || type.m_ctype==CT_VECTOR ||
+					type.m_ctype==CT_SGMATRIX || type.m_ctype==CT_MATRIX)
 				{
 					SGVector<float64_t> value(*((float64_t **)param->m_parameter),
-							(*type.m_length_y));
+							type.get_num_elements(), false);
```

Comment:
@karlnapf
The previous code has a bug. The false must be used to avoid multiple deletion. The bug did not be detected since in previous examples, parameters are scalars. The bug did appear when GaussianARDKernel is used since the weights parameter is not scalar.